### DISCUSSION TITLE
Other object detection

### DISCUSSION THREAD

**Discussion by kbmetz** [blakeblackshear/frigate#857] [General]
From the docs it appears you can only track people and car's currently. Is there any option to include animals?

Also, I've read in regards to object detection that you can 'train' different models. Is it possible to keep track of objects. For example, when it recognizes a car its saved as 'car 1', and the second car 'car 2', and so on?

Just started using Frigate and its pretty amazing. Only 2 camera's but the object detection is really nothing short of amazing compared to the old 'motion detection' of camera's.

**Comment by paularmstrong** [comment]
Seems like we lost some info from the docs on where to find what's available. For now, you can track anything that's in the [coco labelmap](https://blakeblackshear.github.io/frigate/configuration/advanced#customizing-the-labelmap). Animals are definitely possible, depending if what you want is in the label map file that's linked

**Reply by kbmetz** [reply]
Perfect, thanks!

**Reply by kbmetz** [reply]
I added in that label file, and tried to have it tracking 'trucks' but after several days I haven't seen anything. Am I missing something?

**Reply by kbmetz** [reply]
Just going to add my own updates here in case it helps anyone else. Turns out don't need the label file, just need to add objects to be tracked. I've been able so far to track a 'bus' and a 'bicycle' but nothing else. It doesn't detect the difference between cars and trucks, and sometimes missed the bus identification. Doesn't work for dogs or any other animals as far as I can tell. I'm guessing maybe there's somewhere I need to 'train' the model

**Reply by hagak** [reply]
Double check the contents of the labelmap.txt actually being used, I am using the builtin one and it does NOT match what is in the docs.  It has 7 mapped to car which is actually truck so if you want to map to truck you need to change the labelmap.txt so that 7 is truck.

edit: just checked the github version and it also has 7 mapped to car.

**Reply by blakeblackshear** [reply]
Truck and car are consistently confused, so I do not recommend switching 7 back to truck. It will cause cars and trucks to generate multiple overlapping events over and over.

**Reply by kbmetz** [reply]
I've never had it recognize a truck. I have had overlapping events for deer. It thought it was a dog, sheep and a cow.

**Reply by hagak** [reply]
That is because the model does not have a deer modeled.  I actually find it odd they have zebra, beer, giraffe modeled but not deer.

**Reply by kbmetz** [reply]
How could I model a deer? Is there any docs on this (I'm probably missing something obvious)

**Comment by McPaHa** [comment]
Is it possible to detect deer or moose/elg? I have coral.ai usb accelerator and like to know is it somehow easy or what is best way to do it?

**Reply by tubalainen** [reply]
Ive tried "Zebra" and it doesnt work on either Elks or Deers. ;)

**Comment by jkmaxwell** [comment]
See https://github.com/blakeblackshear/frigate/discussions/1043
